llmfoundry/data/finetuning/tasks.py

```diff
@@ -774,9 +771,12 @@ def build_from_hf(
                 if not all(
                         Path(f).suffix in SUPPORTED_EXTENSIONS
                         for f in dataset_files):
-                    raise ValueError(
-                        f'Dataset at local path {dataset_name} contains invalid file types. '
-                        + f'Allowed file types are: {SUPPORTED_EXTENSIONS}')
+                    raise InvalidFileExtensionError(dataset_name,
+                                                    SUPPORTED_EXTENSIONS)
+
+            if split is None:
```

Comment:
any particular reason you added this? The split none check is higher up in the stack (dataloader.py)

Reply:
I was running a manual test with a run that was missing an HF split and was finding that in `_build_collate_fn` before the split check later in the ft dataloader, we end up calling `build_from_hf` which would throw the missing split exception.

Reply:
hm, I don't think is quite the correct condition. If you look where its checked in the dataloader, split is only required if its a remote jsonl dataset, not necessarily required if its an actual hf dataset. at least that is the idea. also the `_build_collate_fn` should go through the same `build_finetuning_dataloader` code, so should hit the other error. How was it failing?

Reply:
I get a 
```
│ /usr/lib/python3/dist-packages/llmfoundry/data/finetuning/tasks.py:793 in    │
│ build_from_hf                                                                │
│                                                                              │
│   790 │   │   │   detected_cpus_with_margin = detected_cpu_count - 8         │
│   791 │   │   │   num_cpus_to_use = max(1, detected_cpus_with_margin)        │
│   792 │   │   │                                                              │
│ ❱ 793 │   │   │   columns_to_remove = list(dataset[0].keys())                │
│   794 │   │   │   tokenized_dataset = dataset.map(                           │
│   795 │   │   │   │   dataset_mapper,                                        │
│   796 │   │   │   │   batched=False,                                         │
│                                                                              │
│ /usr/lib/python3/dist-packages/datasets/dataset_dict.py:80 in __getitem__    │
│                                                                              │
│     77 │   │   │   │   split for split in (Split.TRAIN, Split.TEST, Split.VA │
│     78 │   │   │   ]                                                         │
│     79 │   │   │   suggested_split = available_suggested_splits[0] if availa │
│ ❱   80 │   │   │   raise KeyError(                                           │
│     81 │   │   │   │   f"Invalid key: {k}. Please first select a split. For  │
│     82 │   │   │   │   f"`my_dataset_dictionary['{suggested_split}'][{k}]`.  │
│     83 │   │   │   │   f"Available splits: {sorted(self)}"                   │
╰──────────────────────────────────────────────────────────────────────────────╯
KeyError: "Invalid key: 0. Please first select a split. For example:
`my_dataset_dictionary['train'][0]`. Available splits: ['test', 'train']"
``` 

when I use the `mosaicml/dolly_hhrlhf` as my hf dataset name with no split which isn't caught by our code?

Reply:
I believe that if the dataset _has_ splits, you must specify one, but I think you can have a dataset without splits. I'm actually not 100% sure about this. Maybe its ok to just force split here. I can't find an example dataset that doesn't have a split right now, so maybe I'm just wrong about that. @alextrott16 do you know?

Reply:
Moved the check up in the dataloader and made the split not option in `build_from_hf`!